articles/tips-and-tricks-for-deep-learning/index.md

```diff
@@ -0,0 +1,252 @@
+
+![Transfer Learning with a medium training size](/engineering-education/tips-and-tricks-for-deep-learning/medium-tl.PNG)<br>
+
+*[Image Source: StanFord](https://stanford.edu/~shervine/teaching/cs-230/cheatsheet-deep-learning-tips-and-tricks)*
+
+If you have a medium training size, perform transfer learning on a neural network where most of the layers with the pre-trained weights, excluding the last layer and the softmax layer, have been frozen as shown above. This gives you a little more room to train your data on. 
+
+![Transfer Learning with a large training size](/engineering-education/tips-and-tricks-for-deep-learning/large-tl.PNG)<br>
+
+*[Image Source: StanFord](https://stanford.edu/~shervine/teaching/cs-230/cheatsheet-deep-learning-tips-and-tricks)*
+
+In cases where you have a large training data, you can leverage training weights on almost all the layers and softmax layer while initializing your weights on the pre-trained data. 
+
+Transfer Learning works well in two fields of Artificial Intelligence:
+1. Computer Vision
+2. Natural Language Processing (NLP)
+
+##### Xavier Initialization
+
+Xavier initialization is also known as the Glorot Initialization. It is a weight initialization technique used to initialize the weights in a deep neural network. This technique helps avoid neuron activation functions, starting in very saturated and dead regions.
+
+There are three points worth noting about this technique:
+
+1. The initialized weights at the start shouldn't be set too small. The signal will propagate within the neural network shrinking as it passes through the layers. In the end, the signal will be too tiny to be of any importance.
+2. On the contrary, the weights at the beginning shouldn't also be set too large. These signals will propagate within the neural network making our output signal to become too large.
+3. The initialized weight should be set just right in the middle. Not too large and too small. Once this is accomplished, it helps prevent the neural network from experiencing both the vanishing and exploding gradient problem.  
+
+Xavier initialization is the default initialization in some frameworks. It is an excellent choice for both the sigmoid and hyperbolic tangent (tanh) activation functions. With regards to the popular ReLU function, He initialization is preferred as it performs poorly on Xavier initialization.
+
+#### Optimizing convergence
+
+Optimization involves finding a set of parameters that minimize or maximize a function. 
+
+##### Learning Rate
+
+Learning Rate is commonly denoted with the alpha hyperparameter. It can be described as the rate at which the weights in a neural network get updated.
+
+###### Adaptive Learning Rates
+
+By allowing the learning rate to vary during training, the training time tends to reduce and improves the results. Several methods set a different learning rate for each trainable parameter and adaptively adjust the learning rates. They include:  
```

Comment:
Include a GIF showing the comparison of convergence of the different algorithms. I remember seeing them on Google images.